### ISSUE TITLE
Can't use ansible (from homebrew) after reinstalling newest version

### ISSUE THREAD

**Issue by jedemenge-it** [ansible/ansible#16200]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### CONFIGURATION

I have vault_password_file set in my ansible.cfg.
##### OS / ENVIRONMENT

Mac OS 10.11.5 / Ansible is installed via Homebrew
##### SUMMARY

I can not run ansible-playbook, always get error "ERROR! Unexpected Exception: 'EntryPoint' object has no attribute 'resolve'"
##### STEPS TO REPRODUCE

```
ansible-playbook servers.yml -vvv
```

##### EXPECTED RESULTS

Runs playbook
##### ACTUAL RESULTS

```
ERROR! Unexpected Exception: 'EntryPoint' object has no attribute 'resolve'
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/bin/ansible-playbook", line 92, in <module>
    exit_code = cli.run()
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/cli/playbook.py", line 132, in run
    inventory = Inventory(loader=loader, variable_manager=variable_manager, host_list=self.options.inventory)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/inventory/__init__.py", line 85, in __init__
    self.parse_inventory(host_list)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/inventory/__init__.py", line 148, in parse_inventory
    host.vars = combine_vars(host.vars, self.get_host_variables(host.name))
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/inventory/__init__.py", line 541, in get_host_variables
    self._vars_per_host[hostname] = self._get_host_variables(hostname, vault_password=vault_password)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/inventory/__init__.py", line 571, in _get_host_variables
    vars = combine_vars(vars, self.get_host_vars(host))
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/inventory/__init__.py", line 703, in get_host_vars
    return self._get_hostgroup_vars(host=host, group=None, new_pb_basedir=new_pb_basedir)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/inventory/__init__.py", line 750, in _get_hostgroup_vars
    results = combine_vars(results, self._variable_manager.add_host_vars_file(base_path, self._loader))
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/vars/__init__.py", line 562, in add_host_vars_file
    (name, data) = self._load_inventory_file(path, loader)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/vars/__init__.py", line 546, in _load_inventory_file
    data = loader.load_from_file(new_path)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/parsing/dataloader.py", line 113, in load_from_file
    (file_data, show_content) = self._get_file_contents(file_name)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/parsing/dataloader.py", line 172, in _get_file_contents
    data = self._vault.decrypt(data)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/parsing/vault/__init__.py", line 169, in decrypt
    b_data = this_cipher.decrypt(b_data, self.b_password)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/parsing/vault/__init__.py", line 679, in decrypt
    key1, key2, iv = self.gen_key_initctr(password, salt)
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/lib/python2.7/site-packages/ansible/parsing/vault/__init__.py", line 621, in gen_key_initctr
    backend = default_backend()
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/vendor/lib/python2.7/site-packages/cryptography/hazmat/backends/__init__.py", line 35, in default_backend
    _default_backend = MultiBackend(_available_backends())
  File "/usr/local/Cellar/ansible/2.1.0.0/libexec/vendor/lib/python2.7/site-packages/cryptography/hazmat/backends/__init__.py", line 22, in _available_backends
    "cryptography.backends"
AttributeError: 'EntryPoint' object has no attribute 'resolve'
```

**Comment by sivel** [comment]
This appears to be an issue with a homebrew vendored Python module. Since we do not support nor recommend installing Ansible via homebrew I'm going to close this out.

The recommended installation method for Ansible on Mac is to do so using pip.

**Comment by sebastiaanluca** [comment]
I just updated my pip installation to 2.1.1.0 and have the same issue.

```
ERROR! Unexpected Exception: 'EntryPoint' object has no attribute 'resolve'
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 92, in <module>
    exit_code = cli.run()
  File "/Library/Python/2.7/site-packages/ansible/cli/playbook.py", line 154, in run
    results = pbex.run()
  File "/Library/Python/2.7/site-packages/ansible/executor/playbook_executor.py", line 74, in run
    self._inventory.set_playbook_basedir(os.path.dirname(playbook_path))
  File "/Library/Python/2.7/site-packages/ansible/inventory/__init__.py", line 709, in set_playbook_basedir
    self.get_group_vars(group)
  File "/Library/Python/2.7/site-packages/ansible/inventory/__init__.py", line 727, in get_group_vars
    return self._get_hostgroup_vars(host=None, group=group, new_pb_basedir=new_pb_basedir, return_results=return_results)
  File "/Library/Python/2.7/site-packages/ansible/inventory/__init__.py", line 788, in _get_hostgroup_vars
    host_results = self._variable_manager.add_group_vars_file(base_path, self._loader)
  File "/Library/Python/2.7/site-packages/ansible/vars/__init__.py", line 616, in add_group_vars_file
    data = self._load_inventory_file(path, loader)
  File "/Library/Python/2.7/site-packages/ansible/vars/__init__.py", line 554, in _load_inventory_file
    results = self._load_inventory_file(path=p, loader=loader)
  File "/Library/Python/2.7/site-packages/ansible/vars/__init__.py", line 569, in _load_inventory_file
    data = loader.load_from_file(path)
  File "/Library/Python/2.7/site-packages/ansible/parsing/dataloader.py", line 113, in load_from_file
    (file_data, show_content) = self._get_file_contents(file_name)
  File "/Library/Python/2.7/site-packages/ansible/parsing/dataloader.py", line 172, in _get_file_contents
    data = self._vault.decrypt(data)
  File "/Library/Python/2.7/site-packages/ansible/parsing/vault/__init__.py", line 179, in decrypt
    b_data = this_cipher.decrypt(b_data, self.b_password)
  File "/Library/Python/2.7/site-packages/ansible/parsing/vault/__init__.py", line 689, in decrypt
    key1, key2, iv = self.gen_key_initctr(password, salt)
  File "/Library/Python/2.7/site-packages/ansible/parsing/vault/__init__.py", line 631, in gen_key_initctr
    backend = default_backend()
  File "/Library/Python/2.7/site-packages/cryptography/hazmat/backends/__init__.py", line 35, in default_backend
    _default_backend = MultiBackend(_available_backends())
  File "/Library/Python/2.7/site-packages/cryptography/hazmat/backends/__init__.py", line 22, in _available_backends
    "cryptography.backends"
AttributeError: 'EntryPoint' object has no attribute 'resolve'
```

**Comment by jcdejong** [comment]
@sebastiaanluca try updating ansible? After I ran `sudo pip install ansible --upgrade` everything worked again for me

**Comment by sebastiaanluca** [comment]
I've lost count of many times I reinstalled Ansible and all its dependencies really :p Probably 30+ the day after I posted that message. Anyhow, "fixed" it by removing any pip/python/… module I could find and reinstalling/upgrading/… them all. Not sure what exactly did it (tried a lot of combinations), but now I just got some crypt warning every time I run Ansible. Works for me, but annoying as hell.
